### PR TITLE
[directx-headers] port updated

### PR DESCRIPTION
**Pull request by walbourn**
The DirectX 12 Agility SDK was released today, and this uses version 1.4.9 of the headers in the DirectX-Headers repo.

> In addition to updating the version, I also converted ``CONTROL`` to ``vcpkg.json``

### CODE REVIEW

File: ports/directx-headers/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "directx-headers",
+  "version-string": "1.4.9",
```
Comment (JonLiu1993) [inline ports/directx-headers/vcpkg.json:3]: @walbourn ,please change "version-string" to "version"

File: versions/d-/directx-headers.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "6cb619972dd6eb257e5416c747dd0a65f301c9b7",
+      "version-string": "1.4.9",
```
Comment (JonLiu1993) [inline versions/d-/directx-headers.json:5]: please change "version-string" to "version"

Comment (walbourn) [reply versions/d-/directx-headers.json:5]: Ah, per [this blog](https://devblogs.microsoft.com/cppblog/take-control-of-your-vcpkg-dependencies-with-versioning-support/). On it.

Comment (walbourn) [reply versions/d-/directx-headers.json:5]: BTW, someone should update [this page](https://github.com/microsoft/vcpkg/blob/master/docs/specifications/manifests.md) for the new version stuff.

**Comment by walbourn** [comment]
This port doesn't support OSX, so no idea why that test is failing.

**Comment by JonLiu1993** [comment]
@walbourn ,This is not a problem of directx-headers itself, it has been solved now, we wait for the result after rerun

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Awesome, thanks for the PR!

**Pull request merged**